assets/questions/aad.js

```diff
@@ -0,0 +1,309 @@
+        "b"
+      ],
+      allowedOutputs: ['ab']
+    }, {
+      input: [
+        "aaec",
+        "aaed",
+        "aac",
+        "abe",
+        "abc",
+        "bed",
+        "bea",
+        "bcc",
+        "bcd"
+      ],
+      allowedOutputs: ['ecdab']
+    }, {
+      input: [
+        "egdi",
+        "egdb",
+        "egej",
+        "egei",
+        "ebjj",
+        "ebjg",
+        "ebgh",
+        "ebgj",
+        "igac",
+        "igae",
+        "igia",
+        "igif",
+        "idcf",
+        "idcc",
+        "idh"
+      ],
+      allowedOutputs: ['afchjgdeib']
+    },
+    {
+      input: [
+        "curwfe",
+        "curwfe",
+        "curwst",
+        "curwsw",
+        "curuyc",
+        "curuyf",
+        "curuga",
+        "curugs",
+        "cuxdeh",
+        "cuxdeu",
+        "cuxdbv",
+        "cuxdbn",
+        "cuxolc",
+        "cuxolb",
+        "cuxojd",
+        "cuxojs",
+        "cpcnow",
+        "cpcnob",
+        "cpcnqu",
+        "cpcnqp",
+        "cpctkj",
+        "cpctkd",
+        "cpctfg",
+        "cpctfa",
+        "cpbjeq",
+        "cpbjec",
+        "cpbjz",
+        "cpbhx",
+        "cpbhc",
+        "khjzs",
+        "khjzu",
+        "khjis",
+        "khjiv",
+        "khfax",
+        "khfay",
+        "khftg",
+        "khftq",
+        "kmima",
+        "kmimd",
+        "kmiej",
+        "kmies",
+        "kmrbm",
+        "kmrbr",
+        "kmrus",
+        "kmrut"
+      ],
+      allowedOutputs: ['ljhmezirxygadoqckfsvntwbup']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Next, double-check your code to make sure it handles unexpected cases',
+        ', such as there\'s no solution to the test case or there\'re more ',
+        'than one answers.'
+      ].join(''),
+      [
+        'For the test case that has no solution, just return "". For the test ',
+        'case that has multi-answers, return the answer has the smallest ',
+        'lexicographic order in English alphabet.'
+      ].join('')
+    ],
+    prerequisiteSkills: ['Topology Sorting', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'find_dict',
+    correctnessTests: [{
+      input: [
+        "a",
+        "b",
+        "a"
+      ],
+      allowedOutputs: ['']
+    }, {
+      input: [
+        "c",
+        "a",
+        "b",
+        "a"
+      ],
+      allowedOutputs: ['']
+    }, {
+      input: [
+        "cab",
+        "bc"
+      ],
+      allowedOutputs: ['acb']
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.wrongInNoSolution',
+      messages: [
+        [
+          'You may try this case : ["c", "a", "b", "a"]'
+        ].join('')
+      ]
+    }],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Last, Let\'s fix the solution to adapt lowercase and uppercase. We ',
```

Comment:
Finally, modify your code to handle strings with a mix of lowercase and uppercase letters. Assume that a lowercase and uppercase version of a letter is the same with regards to lexicographic ordering. Your answer, though, should be lowercase.